File: Java/src/main/java/knowledge_layer/TacticRepository.java
package knowledge_layer;

import specification.AdaptationTactic;

import java.util.ArrayList;

public class TacticRepository {
    private ArrayList<AdaptationTactic> tactics;

    public TacticRepository(ArrayList<AdaptationTactic> adaptationTactics) {
        tactics = adaptationTactics;
    }

    public ArrayList<AdaptationTactic> getPossibleTactics() {
        return tactics;
    }

    public void addTactic(AdaptationTactic adaptationTactic) {
        tactics.add(adaptationTactic);
    }
}

File: Python/knowledge_layer/TacticRepository.py
from typing import List

from specification.AdaptationTactic import AdaptationTactic


class TacticRepository:

    def __init__(self, adaptationTactics: List[AdaptationTactic]):
        self._tactics = adaptationTactics

    def getPossibleTactics(self) -> List[AdaptationTactic]:
        return self._tactics

    def addTactic(self, adaptationTactic: AdaptationTactic):
        self._tactics.append(adaptationTactic)
